Content.Server/GameObjects/Components/Chemistry/PourableComponent.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Content.Server.GameObjects.Components.Nutrition;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Chemistry
+{
+    /// <summary>
+    /// Gives an entity click behavior for pouring reagents into
+    /// other entities and being poured into. The entity must have
+    /// a SolutionComponent or DrinkComponent for this to work.
+    /// (DrinkComponent adds a SolutionComponent if one isn't present).
+    /// </summary>
+    [RegisterComponent]
+    [ComponentReference(typeof(IAttackBy))]
```

Comment:
You don't actually have to register `IAttackBy`.

Reply:
Alright, removed that. When would I need to register it?